### PR TITLE
Instruments pgevents to report metrics

### PR DESCRIPTION
**Pull request by hugoShaka**
Most event backends provide no or few metrics, I needed to measure those when testing the cockroachDB event backend so here they are.

Once you are happy with this PR I can send similar ones for most popular event backends (especially DyamoDB).

I copied most of the logic from the existing firestore metrics with the following changes:
- ~I turned `batchWrite` into `cleanup` as this was the only use case and found that `batchWrite` might be confusing. Maybe `batchDelete` is more appropriate, I'll take any suggestion~ switched to batchDelete.
- I added a `status: success|failure` label to the counters because we had no visibility on the request failure (are we failing the call? getting rate-limited?). When observing a system it's a good practice to measure at least [Rate Errors and Duration](https://grafana.com/blog/2018/08/02/the-red-method-how-to-instrument-your-services/), so this PR adds the "Errors" part.

Changelog: add Prometheus metrics for the Postgres event backend.

### CODE REVIEW

**Review by espadolini — APPROVED** [review]

File: lib/events/pgevents/metrics.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			Namespace: teleport.MetricNamespace,
+			Name:      "postgres_events_backend_batch_read_requests",
+			Help:      "Number of batch read requests to postgres events",
+		},
+		labels,
+	)
+	cleanupRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      "postgres_events_backend_cleanup_requests",
```
[the remark this answers is not in the document]
Comment (hugoShaka) [reply lib/events/pgevents/metrics.go:52]: It's the query we run every X to cleanup events older than the retention period. I will rename batchDelete, I think it's clearer for the user what we're doing.

[the remark this answers is not in the document]
Comment (espadolini) [reply lib/events/pgevents/metrics.go:52]: +1 for "batch delete"

**Review by rosstimothy — COMMENTED** [review]
Please also update the metrics reference docs: https://goteleport.com/docs/reference/metrics

**Review by rosstimothy — APPROVED** [review]
Thanks Hugo!

**Pull request merged**